classes/submission/SubmissionDAO.inc.php

```diff
@@ -111,10 +112,104 @@ function _fromRow($row) {
 		$submission->setLanguage($row['language']);
 		$submission->setCommentsToEditor($row['comments_to_ed']);
 
-		$this->getDataObjectSettings('submission_settings', 'submission_id', $submission->getId(), $submission);
+		$this->getSubmissionSettings($submission, $submissionRevision);
 
 		return $submission;
 	}
+	
+	/**
+	 * Get contents of the submission_settings table
+	 * @param $submission Submission object
+	 * @param $submissionRevision int optional
+	 */
+	function getSubmissionSettings($submission, $submissionRevision = null) {
+
+		if ($submissionRevision == null) {
+			$sql = "SELECT * FROM submission_settings WHERE submission_id = ?";
+			$params = array($submission->getId());
+		} else {
+			$sql = "SELECT * FROM submission_settings WHERE submission_id = ? AND submission_revision = ?";
+			$params = array($submission->getId(), $submissionRevision);
+		}
+		$result = $this->retrieve($sql, $params);
```

Comment:
Consider writing this block as a single SQL statement, e.g.:

    $params = array((int) $submission->getId());
    if ($submissionRevision) $params[] = (int) $submissionRevision;
    $result = $this->retrieve(
        'SELECT * FROM submission_settings WHERE submission_id = ?' .
        ($submissionRevision?' AND submission_revision = ?':''), 
        $params
    );

(Also note the added int cast for submission revision.)

This duplicates less code and if we later had to add another potential condition it would scale better.